core/store/models/eth.go

```diff
@@ -190,16 +192,13 @@ func (f FunctionSelector) WithoutPrefix() string { return f.String()[2:] }
 // SetBytes sets the FunctionSelector to that of the given bytes (will trim).
 func (f *FunctionSelector) SetBytes(b []byte) { copy(f[:], b[:FunctionSelectorLength]) }
 
-// UnmarshalJSON parses the raw FunctionSelector and sets the FunctionSelector
-// type to the given input.
-func (f *FunctionSelector) UnmarshalJSON(input []byte) error {
-	var s string
-	err := json.Unmarshal(input, &s)
-	if err != nil {
-		return err
-	}
+var hexRegexp *regexp.Regexp = regexp.MustCompile("^[0-9a-fA-F]*$")
 
+func unmarshalFromString(s string, f *FunctionSelector) error {
 	if utils.HasHexPrefix(s) {
```

Comment:
Why aren't we using a consistent representation here?

Reply:
It is a consistent representation. I just pulled this function out to appease Code Climate's 4-`return` rule.

Reply:
I mean that it can either start with a hash prefix or it cannot, why support two encodings?

Reply:
Being able to represent the selector as both its hash prefix and the method signature it encodes seems very convenient.